switch/switch_main.ml

```diff
@@ -176,9 +176,17 @@ let make_server config =
       >>= fun qs ->
       on_disk_queues := qs;
       info "Reading the redo-log from %s" redo_log_path;
-      let open Lwt_result in
-      Block.connect ("buffered:" ^ redo_log_path)
+      Block.connect ~buffered:true redo_log_path
+      >>= fun block -> begin
+        match block with
+        | `Ok block -> Lwt.return block
+        | `Error e ->
```

Comment:
Yes. But unless we have a PPX macro, we need to provide __LOC__ from the error location and can't abstract it away like in `debug` and `info`.